sled-agent/src/services.rs

```diff
@@ -2754,6 +2758,10 @@ impl ServiceManager {
                                 &format!("[{address}]:{MGS_PORT}"),
                             )?;
 
+                            if let Some(info) = self.inner.sled_info.get() {
```

Comment:
The fact that this is optional makes me slightly nervous - the underlay `address` is _not_ optional at this point, so at least per the types we could have an underlay addr but no rack ID. Is that actually possible in practice?

Reply:
I'm not sure if it's possible or not, but I suspect it is not possible in practice. The bootstrap server must have received a `StartSledAgentRequest` for it to start sled agent and learn about both the underlay network and the rack_id. It appears that the underlay address is only configured in [SledAgentInner](https://github.com/oxidecomputer/omicron/blob/main/sled-agent/src/bootstrap/server.rs#L289-L322) from the `StartSledAgentRequest`, and that this happens after [`sled_agent_started` is called with the `rack_id `](https://github.com/oxidecomputer/omicron/blob/main/sled-agent/src/sled_agent.rs#L416-L422) from the same request. 

In the cold boot case, the underlay address is only touched after the `SledAgentRequest` is [loaded](https://github.com/oxidecomputer/omicron/blob/main/sled-agent/src/bootstrap/server.rs#L290-L338).

I'm not really sure what we should do here. I can certainly call `unwrap` to indicate a programmer error, or just log an error, which will in the end result in not being able to add sleds, but also not crash existing working code.

Reply:
I don't think I'd `unwrap()`. I'd definitely leave a comment as a breadcrumb for any future cleanup work, and I'd be fine with or without an error/warn log.

Reply:
Added in [347b378](https://github.com/oxidecomputer/omicron/pull/4145/commits/347b378a8dbfb860f0ce7e183abafcd7af9fefdb)